### PR TITLE
White glove: Show offer page

### PR DESCRIPTION
**Pull request by niranjan-uma-shankar**
#### Changes proposed in this Pull Request

This PR is a part of the larger PR that implements the white glove A/B test. Please check https://github.com/Automattic/wp-calypso/pull/41618 for more context.

* This PR implements the white glove offer page, and shows the offer to users who select a free plan in signup. 
* Note that this PR is only for implementing the offer page, so the Buy Now CTA will add only a Business plan for now(instead of a "white glove" product).

#### Testing instructions

**Scenario 1** (verify A/B test assignment works)

1. Set your locale to EN and go through the signup flow. by selecting a free plan and free domain.
2. You'll either land in Customer Home or see the offer page. In both cases, check that the AB test assignment has put you either in control or variation. Type `localstorage.ABTests;` in your browser console, you should see an entry for `whiteGloveUpsell `.
3. Set your locale to non-EN and go through signup flow. Verify that you don't get assigned to the test if you select a free plan and free domain.

**Scenario 2** (verify offer page UI)

1. Put yourself in the variant of the `whiteGloveUpsell` AB test and go through the signup flow.
2. Select a free plan and free domain and verify that you see the offer page.
3. Verify that the UI checks out with the [Figma](https://www.figma.com/file/G52TF9nObKGWqwPfDx3Wym/White-Gloves-Interstitial?node-id=50%3A307).
4. Verifythat the UI checks out in mobile screen sizes.

### CODE REVIEW

**Comment by ballio2000** [comment]
> @ballio2000 I've implemented the UI feedback comments you mentioned. So, cc'ing you once again to confirm everything looks good for you.

Looking good! Thanks @niranjan-uma-shankar! We can further tweak and polish if it tests well! :)

File: client/my-sites/checkout/upsell-nudge/white-glove/index.jsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+ * Image dependencies
+ */
+
+export class WhiteGlove extends PureComponent {
+	render() {
+		const title = 'Checkout ‹ White Glove';
+
+		return (
+			<>
+				<PageViewTracker path="/checkout/offer-white-glove/:site" title={ title } />
```
Comment (travisw) [inline client/my-sites/checkout/upsell-nudge/white-glove/index.jsx:31]: Just noting that this is different from the actual URL path which would be `/checkout/:site/offer-white-glove`.
It looks like the other offer pages don't use either order consistently.

**Comment by niranjan-uma-shankar** [comment]
Thanks for the review!

> The only different with Figma was $299 in Figma compared to $300 on the actual page.

I brought it up in pau2Xa-13X-p2#comment-4997. We will have to leave it as $300 for this test. I tried exploring if we could auto-apply a coupon but the minimum discount value for a coupon is 1%(so $3) so that wouldn't work out.

> On smaller mobile screens the "buy now" button could probably have it's left/right padding reduced a bit.
> Just noting that this is different from the actual URL path which would be /checkout/:site/offer-white-glove.

I've pushed in a fix for both.

**Comment by travisw** [comment]
Cool, looks good. 👍

**Pull request merged**